editor/src/document/document_message_handler.rs

```diff
@@ -43,6 +44,52 @@ impl DocumentsMessageHandler {
 	pub fn active_document_mut(&mut self) -> &mut DocumentMessageHandler {
 		&mut self.documents[self.active_document_index]
 	}
+	fn get_new_doc_title(&self) -> String {
+		let digits = ('0'..='9').collect::<Vec<char>>();
+		let mut doc_title_numbers = self
+			.documents
+			.iter()
+			.map(|d| {
+				if d.name.ends_with(digits.as_slice()) {
+					let (_, number) = d.name.split_at(17);
```

Comment:
So it looks like there are multiple concerns with this existing logic (this PR just pulls it into its own function). Is this PR the right place to upgrade it or should that be done separately?

Reply:
I suppose a natural consequence of it showing up in the diff is that it's fair game for code review. You, and all contributors, are always welcome to file an issue or write a TODO comment indicating that it'll should be done later if you would prefer to hurry up and get the main feature merged. That said, this looks like a moderately quick cleanup of existing code and we encourage cleaning up dirty laundry whenever you see it (as long as it doesn't involve a considerably large refactor that moves lots of code which is unrelated to the original feature, then it would be preferable to work on the two things in separate branches). But when there are strict procedures like "don't change touch anything unrelated to your current issue!" then it's unlikely that the code ever gets cleaner throughout use. So please do clean things up wherever you see something that could be improved— document any code that you stared at for a while in confusion before learning how it works, rename any undescriptive/vague/ambiguous/abbreviated variable/function/type names, move a function to a more relevant file, break out complex code into simpler functions, etc.

Reply:
> So it looks like there are multiple concerns with this existing logic (this PR just pulls it into its own function). Is this PR the right place to upgrade it or should that be done separately?

Some changes like this one:
```rust
let (new_num, _) = doc_title_numbers.enumerate().find(|(i, number)| i != number).unwrap_or(self.documents.len());
```

are purely cosmetic and you are always welcome to fix these, especially if it is a relatively quick fix  and a solution is already provided.

Something like the magic number 17 actually requires changes because your pr would otherwise break the code.
Fixes that fix pieces of code your current pr breaks should always be done in the same pr